sugondat/shim/src/cmd/query/mod.rs

```diff
@@ -16,6 +16,31 @@ pub async fn run(params: Params) -> anyhow::Result<()> {
     Ok(())
 }
 
+pub async fn get_block_at(
+    client: &sugondat_rpc::Client,
+    block: BlockParams,
+) -> anyhow::Result<sugondat_rpc::Block> {
+    let maybe_hash = match block.block_ref {
+        BlockRef::Best => None,
+        BlockRef::Hash(h) => Some(h),
+        BlockRef::Number(n) => Some(match block.wait {
+            true => client.wait_finalized_height(n).await,
```

Comment:
Food for thought: why finalized?

Reply:
That's true, I think I used 'finalized' there just because that method was already present. However, this command does not wait until the block is finalized, but simply waits until it is available. 

I will modify this and add a function called 'wait_block_hash'

Reply:
it's ok to leave it for a follow up. Hence food for thought.